Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitInspector.cs

```diff
@@ -51,41 +54,45 @@ public override void OnInspectorGUI()
             EditorGUILayout.PropertyField(activeProfile);
             bool changed = EditorGUI.EndChangeCheck();
             string commandName = Event.current.commandName;
-            var allConfigProfiles = ScriptableObjectExtensions.GetAllInstances<MixedRealityToolkitConfigurationProfile>();
 
-            if (activeProfile.objectReferenceValue == null && currentPickerWindow == -1 && checkChange && !BuildPipeline.isBuildingPlayer)
+            // If not profile is assigned, then warn user
+            if (activeProfile.objectReferenceValue == null)
             {
-                if (allConfigProfiles.Length > 1)
+                EditorGUILayout.HelpBox("MixedRealityToolkit cannot initialize unless an Active Profile is assigned!", MessageType.Error);
+
+                if (GUILayout.Button("Assign MixedRealityToolkit Profile") || forceShowProfilePicker)
                 {
-                    EditorUtility.DisplayDialog("Attention!", "You must choose a profile for the Mixed Reality Toolkit.", "OK");
-                    currentPickerWindow = GUIUtility.GetControlID(FocusType.Passive);
+                    forceShowProfilePicker = false;
+
+                    var allConfigProfiles = ScriptableObjectExtensions.GetAllInstances<MixedRealityToolkitConfigurationProfile>();
+
                     // Shows the list of MixedRealityToolkitConfigurationProfiles in our project,
                     // selecting the default profile by default (if it exists).
-                    EditorGUIUtility.ShowObjectPicker<MixedRealityToolkitConfigurationProfile>(MixedRealityInspectorUtility.GetDefaultConfigProfile(allConfigProfiles), false, string.Empty, currentPickerWindow);
-                }
-                else if (allConfigProfiles.Length == 1)
-                {
-                    activeProfile.objectReferenceValue = allConfigProfiles[0];
-                    changed = true;
-                    Selection.activeObject = allConfigProfiles[0];
-                    EditorGUIUtility.PingObject(allConfigProfiles[0]);
-                }
-                else
-                {
-                    if (EditorUtility.DisplayDialog("Attention!", "No profiles were found for the Mixed Reality Toolkit.\n\n" +
-                                                                  "Would you like to create one now?", "OK", "Later"))
+                    if (allConfigProfiles.Length > 1)
```

Comment:
>> 1) [](start = 49, length = 4)

Should this be > 0?

Note that this section originally looked for 1 because in that case, we shouldn't prompt (we should just auto-select, since that's the only thing the user can do anyway).

Since we now include the HL2 and HL1 profiles (in addition to the default one) this probably won't get exercised, but changing it to check for empty is probably the more correct thing

Reply:
Yes it should just check for empty